src/MICore/Transports/LocalUnixTransport.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace MICore
+{
+    public class LocalUnixTransport : StreamTransport
+    {
+        private string _dbgStdInName;
+        private string _dbgStdOutName;
+        private FileSystemWatcher _fifoWatcher;
+
+        public override void InitStreams(LaunchOptions options, out StreamReader reader, out StreamWriter writer)
+        {
+            LocalLaunchOptions localOptions = (LocalLaunchOptions)options;
+
+            // Default working directory is next to the app
+            string debuggeeDir;
+            if (Path.IsPathRooted(options.ExePath) && File.Exists(options.ExePath))
+            {
+                debuggeeDir = System.IO.Path.GetDirectoryName(options.ExePath);
+            }
+            else
+            {
+                // If we don't know where the app is, default to HOME, and if we somehow can't get that, go with the root directory.
+                debuggeeDir = Environment.GetEnvironmentVariable("HOME");
+                if (string.IsNullOrEmpty(debuggeeDir))
+                    debuggeeDir = "/";
+            }
+
+            _dbgStdInName = UnixUtilities.MakeFifo(Logger);
+            _dbgStdOutName = UnixUtilities.MakeFifo(Logger);
+
+            string watcherDirectory = _dbgStdInName.Substring(0, _dbgStdInName.IndexOf(UnixUtilities.FifoPrefix, StringComparison.OrdinalIgnoreCase));
+            _fifoWatcher = new FileSystemWatcher(watcherDirectory, UnixUtilities.FifoPrefix + "*");
+            _fifoWatcher.Deleted += FifoWatcher_Deleted;
+            _fifoWatcher.EnableRaisingEvents = true;
```

Comment:
Agreed. @jacdavis specifically is referring to `FileSystemWatcher` on OS X on mono.